src/Core/Prefab/GlobeView.js

```diff
@@ -256,19 +262,24 @@ function GlobeView(viewerDiv, coordCarto, options = {}) {
         v.setFromMatrixPosition(wgs84TileLayer.object3d.matrixWorld);
         var len = v.distanceTo(this.camera.camera3D.position);
         v.setFromMatrixScale(wgs84TileLayer.object3d.matrixWorld);
-        var lim = v.x * size * 1.1;
 
         // TODO: may be move in camera update
         // Compute fog distance, this function makes it possible to have a shorter distance
         // when the camera approaches the ground
         this.fogDistance = mfogDistance * Math.pow((len - size * 0.99) * 0.25 / size, 1.5);
 
-        if (len < lim) {
-            var t = Math.pow(Math.cos((lim - len) / (lim - v.x * size * 0.9981) * Math.PI * 0.5), 1.5);
-            var color = new THREE.Color(0x93d5f8);
-            renderer.setClearColor(color.multiplyScalar(1.0 - t), renderer.getClearAlpha());
-        } else if (len >= lim) {
-            renderer.setClearColor(0x030508, renderer.getClearAlpha());
+        // get altitude camera
+        coordCam.set('EPSG:4978', this.camera.camera3D.position).as('EPSG:4326', coordGeoCam);
+        const altitude = coordGeoCam.altitude();
+
+        // if altitude camera is below a threshold (limitAlti)
+        // to fade the color of space to the color of the sky proportional to the normalized interval [ground, threshold]
+        if (altitude < limitAlti) {
+            const t = (limitAlti - altitude) / limitAlti;
+            const colorSky = spaceColor.clone().lerp(skyBaseColor, t);
```

Comment:
Do we really need to clone each time ? What about adding a global `const skyColor` and `copy()` inside it each time ?

Reply:
yes